[quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30553 [quant][graphmode] `insert_quant_dequant` pass support shared class types
* #30909 [quant][graphmode] Remove `fold_prepack` for now
* **#30860 [quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`**
* #30859 [quant][graphmode][refactor] getQParams return a dictionary of qparams

Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18849021](https://our.internmc.facebook.com/intern/diff/D18849021)

jerryzh168 committed:
[quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`

Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

lly-zero-one commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -738,9 +743,51 @@ c10::optional<script::Module> InsertQuantDeQuantHelper::findChildModuleToQuantiz
+  for (auto& invoked_methods: getInvokedMethods(module, method_name)) {
+    auto& invoked_module = std::get<0>(invoked_methods);
+    const auto& invoked_method_name = std::get<1>(invoked_methods);
+    run(invoked_module, invoked_method_name);
```
Can we push this recursion back to the calling side or getInvokedmethods?

jerryzh168 replied:
You mean call `run` inside `getInvokedMethods`? I feel it's better to structure this way since we split the analysis and transformation. We use the same thing in `InsertObserversHelper`

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18849021](https://our.internmc.facebook.com/intern/diff/D18849021)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Factor out getInvokedMethod in `InsertQuantDeQuantHelper`"


Summary:
att

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18849021](https://our.internmc.facebook.com/intern/diff/D18849021)

[ghstack-poisoned]
